### PR TITLE
fix(crd): bump controller-gen to 0.4.1

### PR DESCRIPTION
**Pull request by greglanthier**
Attempts to address comments in issue #258.

Our team uses this operator to manage connections to IBM Cloud hosted databases.  We will be moving to OpenShift 4.9 / Kubernetes 1.22 in the near future and we would like to continue using this operator.

[This Red Hat OpenShift blog post](https://connect.redhat.com/blog/api-deprecation-kubernetes-122-will-impact-your-operators) provides some recommendations on how to update existing operators to support the move to Kubernetes 1.22.  

This PR bumps the controller-gen version to 0.4.1 and regenerates the CRDs using the `apiextensions.k8s.io/v1` group.

This PR also does the following:
- ignore linter warnings re: the deprecated `controller-runtime/pkg/client/fake` class.
- tweak some OLM-related generation tools
- adds a `minKubeVersion: 1.16.0` field to the generated `ClusterServiceVersion` spec for the operator
- ignore gosec G304 warning (appears to be same category as G307) in a generation tool

Signed-off-by: Greg Lanthier <greg.lanthier@gmail.com>

### CODE REVIEW

File: internal/cmd/genolm/main.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func run(output, repoRoot, versionStr string) error {
 }
 
 func writeFile(path string, tmpl *template.Template, data interface{}) error {
-	f, err := os.Create(path)
+	f, err := os.Create(path) // #nosec G304
```
Comment (JohnStarich) [inline internal/cmd/genolm/main.go:150]: Can you add a justification to this nosec? Alternatively we could remove the nosec – it looks like this one isn't being flagged with a high enough severity when I run gosec locally.

**Review by JohnStarich — CHANGES_REQUESTED** [review]
Thanks for opening, Greg! ❤️ 
Code changes look pretty good, just had 1 comment.

Have you run through an installation of the old version and upgraded to this one? I'll test this out as well, but would like to confirm.

**Comment by JohnStarich** [comment]
After these tweaks to genolm:
```diff
```

And learning the new-new test deployment system for openshift (happens every time somehow 😛), I'm hitting this new error:
image

I think this is a result of IBM Cloud Operator's more flexible schema conflicting with the strict typing system used in CRD v1. Looks like we'll need to dig in a bit more.

File: internal/cmd/genolm/main.go (hunk 1 above)
Comment (greglanthier) [reply internal/cmd/genolm/main.go]: **gosec 2.10.0** was installed on my machine when I originally ran `make lint`.

When I uninstalled **gosec** and re-ran `make lint` the tests resorted to using **gosec 2.9.2** as defined in the Makefile and the lint passed.  I'll amend my PR to remove the `// nosec G304`.

**Comment by JohnStarich** [comment]
Merged via #267

**Pull request closed without merging**